examples/swift/hello_world/AsyncDemoFilter.swift

```diff
@@ -0,0 +1,52 @@
+import Envoy
+import Foundation
+
+class AsyncDemoFilter: AsyncResponseFilter {
+  private var internalCallbacks: ResponseFilterCallbacks?
+  private lazy var callbacks: ResponseFilterCallbacks = {
+    internalCallbacks! //swiftlint:disable:this force_unwrapping
+  }()
+
+  func onResponseHeaders(_ headers: ResponseHeaders, endStream: Bool)
+    -> FilterHeadersStatus<ResponseHeaders>
+  {
+    return .stopIteration
+  }
+
+  func onResponseData(_ body: Data, endStream: Bool) -> FilterDataStatus<ResponseHeaders> {
+    if endStream {
```

Comment:
Might be nice to add inline comments to explain what it's doing (since these serve as demos for readers looking to use the library)